pkg/server/commands/listusers/list_users_rpc_test.go

```diff
@@ -1447,6 +1447,243 @@ func TestListUsersEdgePruning(t *testing.T) {
+		    schema 1.1
+		  type user
+		  type document
+		    relations
+		      define viewer: [user:*,user]
+		      define required: [user:*,user]
+		      define can_view: viewer and required`,
+
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "viewer", "user:*"),
+				tuple.NewTupleKey("document:1", "required", "user:*"),
+				tuple.NewTupleKey("document:1", "viewer", "user:will"),
+				tuple.NewTupleKey("document:1", "required", "user:will"),
+			},
+			expectedUsers: []string{"user:*", "user:will"},
+		},
+		{
+			name: "with_multiple_wildcards_4",
+			req: &openfgav1.ListUsersRequest{
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "can_view",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type: "user",
+					},
+				},
+			},
+			model: `model
+		    schema 1.1
+		  type user
+		  type document
+		    relations
+		      define required_1: [user]
+		      define required_2: [user]
+		      define can_view: required_1 and required_2`,
+
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "required_1", "user:*"),
+				tuple.NewTupleKey("document:1", "required_2", "user:*"),
+
+				tuple.NewTupleKey("document:1", "required_1", "user:maria"),
+				tuple.NewTupleKey("document:1", "required_2", "user:maria"),
+			},
+			expectedUsers: []string{"user:maria"},
+		},
+
+		{
+			name: "with_multiple_wildcards_5",
+			req: &openfgav1.ListUsersRequest{
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "can_view",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type: "user",
+					},
+				},
+			},
+			model: `model
+		    schema 1.1
+		  type user
+		  type document
+		    relations
+		      define required_1: [user]
+		      define required_2: [user:*]
+		      define can_view: required_1 and required_2`,
+
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "required_1", "user:maria"),
+				tuple.NewTupleKey("document:1", "required_2", "user:*"),
+			},
+			expectedUsers: []string{"user:maria"},
+		},
+		{
+			name: "with_multiple_wildcards_6",
+			req: &openfgav1.ListUsersRequest{
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "can_view",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type: "user",
+					},
+				},
+			},
+			model: `model
+		    schema 1.1
+		  type user
+		  type document
+		    relations
+		      define required_1: [user]
+		      define required_2: [user]
+		      define can_view: required_1 and required_2`,
+
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "required_1", "user:maria"),
+				tuple.NewTupleKey("document:1", "required_1", "user:jon"),
+
+				tuple.NewTupleKey("document:1", "required_1", "user:will"),
+				tuple.NewTupleKey("document:1", "required_2", "user:will"),
+
+				tuple.NewTupleKey("document:1", "required_1", "user:*"),
+				tuple.NewTupleKey("document:1", "required_2", "user:*"),
```

Comment:
these two tuples aren't allowed by the model you defined

Reply:
Discussed in similar comment below.

Reply:
Please leave a comment next to each invalid tuple.